### PR TITLE
Cfs 681 introduce step validate

### PR DESCRIPTION
**Pull request by aplarge**
Change transform functions to dataops functions which are passed a rule type to process
Refactored the functions that run sparql queries

### CODE REVIEW

File: ekglib/sparql/sparql_endpoint.py [hunk 1]
```diff
@@ -169,7 +169,7 @@ def execute_sparql_select_query(self, sparql_statement, mime_type=MIME_CSV):
             log_item("Executing", sparql_statement)
         # noinspection PyProtectedMember
         log_item("Statement Type", self.sparql_endpoint._parseQueryType(sparql_statement))
-
+        self.sparql_endpoint.clearCustomHttpHeader("Accept")
```
Comment (szaniszlo) [inline ekglib/sparql/sparql_endpoint.py:172]: ideally sparql_endpoint should be renamed to "request" and we should not be setting this header when creating the request, so then there would be no need to clear it here...

File: tests/test_data/dataops/generic/00001-check-dataset-not-empty.ttl [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                                                         rdfs:label "Check dataset is not empty" ;
+                                                         <https://ekgf.org/ontology/dataset/dataSourceCode> "ldap" ;
+                                                         rule:createsProvenance false ;
+                                                             rule:sparqlQueryType     rule:SPARQLSelectQuery ;
+                                                         rule:definedIn <http://localhost/id/story-validate-rule-file> ;
+#                                                         rule:hasSPARQLRule """PREFIX legal-entity:  <https://ekgf.org/ontology/legal-entity/>
+##ASK {
+##    ?s ?p ?o .
+##}
+##""" ;
```
Comment (szaniszlo) [inline tests/test_data/dataops/generic/00001-check-dataset-not-empty.ttl]: remove?

File: tests/test_data/dataops/generic/00001-check-dataset-not-empty.ttl [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                                                                                              <https://ekgf.org/ontology/dataset/dataSourceCode> "ldap" ;
+                                                                                              rule:createsProvenance false ;
+                                                                                                  rule:sparqlQueryType     rule:SPARQLConstructQuery ;
+                                                                                              rule:definedIn <http://localhost/id/sha256:b8d8bf3f99515add820eab0d75837753ab8a35fc33c3990a1e1c520982c04fe5> ;
+#                                                                                              rule:hasSPARQLRule """PREFIX le:  <https://ekgf.org/ontology/legal-entity/>
+#  INSERT DATA  {
+#    le:subject le:predicate le:object;
+#        le:predicate2 le:object2.
+#  }
+#""" ;
```
Comment (szaniszlo) [inline tests/test_data/dataops/generic/00001-check-dataset-not-empty.ttl]: remove?

File: tests/test_ldap_parser.py [hunk 4]
```diff
@@ -68,38 +68,37 @@ def test_export_from_d_trust_dot_de(test_data_dir, kgiri_base):
...
+#                         kgiri_base=kgiri_base,
+#                         verbose=False
+#                     )
+#             ):
+#                 assert 0 == ldap_parser_to_file()
+#         print('Last 8 lines of output:')
+#         with open(output_file_name, 'r') as file:
+#             for line in (file.readlines()[-8:]):
+#                 print(line, end='')
+#
```
Comment (szaniszlo) [inline tests/test_ldap_parser.py:101]: is this now broken? This used to run, can we resurrect please?

**Review by szaniszlo — APPROVED** [review]

File: tests/test_ldap_parser.py (hunk 4 above)
Comment (aplarge) [reply tests/test_ldap_parser.py:101]: This endpoint no longer seems to be available. I have left the test in incase it should re-emerge.

File: tests/test_data/dataops/generic/00001-check-dataset-not-empty.ttl (hunk 2 above)
Comment (aplarge) [reply tests/test_data/dataops/generic/00001-check-dataset-not-empty.ttl]: Done

File: tests/test_data/dataops/generic/00001-check-dataset-not-empty.ttl (hunk 3 above)
Comment (aplarge) [reply tests/test_data/dataops/generic/00001-check-dataset-not-empty.ttl]: Done

**Pull request merged**